pkg/manifests/external_dns.go

```diff
@@ -54,41 +87,50 @@ func (p Provider) String() string {
 	}
 }
 
-func (p Provider) ResourceName() string {
-	switch p {
+// ExternalDnsConfig defines configuration options for required resources for external dns
+type ExternalDnsConfig struct {
+	TenantId, Subscription, ResourceGroup,
+	ClientId, ServiceAccountName, Namespace string
+	IdentityType       IdentityType
+	ResourceType       ResourceType
```

Comment:
we should support multiple resource types

Reply:
done